client/my-sites/stats/features/modules/stats-locations/stats-locations.tsx

```diff
@@ -50,40 +57,46 @@ interface StatsModuleLocationsProps {
 const StatsLocations: React.FC< StatsModuleLocationsProps > = ( { query, summaryUrl } ) => {
 	const translate = useTranslate();
 	const siteId = useSelector( getSelectedSiteId ) as number;
-	const statType = 'statsCountryViews';
+	const statType = STAT_TYPE_COUNTRY_VIEWS;
 	const isOdysseyStats = config.isEnabled( 'is_running_in_jetpack_site' );
 	const supportUrl = isOdysseyStats ? JETPACK_SUPPORT_URL_TRAFFIC : SUPPORT_URL;
 
-	// Use StatsModule to display paywall upsell.
-	const shouldGateStatsModule = useShouldGateStats( statType );
-
 	const [ selectedOption, setSelectedOption ] = useState( OPTION_KEYS.COUNTRIES );
 	const optionLabels = {
 		[ OPTION_KEYS.COUNTRIES ]: {
 			selectLabel: translate( 'Countries' ),
 			headerLabel: translate( 'Top countries' ),
 			analyticsId: 'countries',
+			feature: STATS_FEATURE_LOCATION_COUNTRY_VIEWS,
 		},
 		[ OPTION_KEYS.REGIONS ]: {
 			selectLabel: translate( 'Regions' ),
 			headerLabel: translate( 'Top regions' ),
 			analyticsId: 'regions',
+			feature: STATS_FEATURE_LOCATION_REGION_VIEWS,
 		},
 		[ OPTION_KEYS.CITIES ]: {
 			selectLabel: translate( 'Cities' ),
 			headerLabel: translate( 'Top cities' ),
 			analyticsId: 'cities',
+			feature: STATS_FEATURE_LOCATION_CITY_VIEWS,
 		},
 	};
 
+	// Use StatsModule to display paywall upsell.
+	const shouldGateStatsModule = useShouldGateStats( statType );
+	const shouldGateTab = useShouldGateStats( optionLabels[ selectedOption ].feature );
+	const shouldGate = shouldGateStatsModule || shouldGateTab;
 	const geoMode = GEO_MODES[ selectedOption ];
 	const title = optionLabels[ selectedOption ]?.selectLabel;
 
 	const {
 		data = [],
 		isLoading: isRequestingData,
 		isError,
-	} = useLocationViewsQuery< StatsLocationViewsData >( siteId, geoMode, query );
+	} = useLocationViewsQuery< StatsLocationViewsData >( siteId, geoMode, query, {
+		enabled: ! shouldGate,
```

Comment:
This is to prevent querying the location-views API if the region/city tab are locked.